.github/actions/restore-node/action.yml

```diff
@@ -21,6 +21,37 @@ runs:
         clean: 'false'
         submodules: 'true'
         persist-credentials: false
+    # Select a branch on Endo to test against by adding text to the body of the
+    # pull request. For example: #endo-branch: some-pr-branch
+    # The default is '*NONE*' to indicate not to check out Endo, just use
+    # the published NPM packages.
+    - name: Get the appropriate Endo branch
+      id: endo-branch
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |
+          let branch = '(use npm packages)';
+          if (context.payload.pull_request) {
+            const { body } = context.payload.pull_request;
+            const regex = /.*\#endo-branch:\s+(\S+)/;
+            const result = regex.exec(body);
+            if (result) {
+              branch = result[1];
+            }
+          }
+          console.log(branch);
+          return branch;
+    - name: check out Endo if necessary
+      uses: actions/checkout@v3
+      with:
+        repository: agoric/endo
+        path: ~/endo
+        ref: ${{ steps.endo-branch.outputs.result }}
+        clean: 'false'
+        submodules: 'true'
+        persist-credentials: false
+      if: '${{ steps.endo-branch.outputs.result }}' != '(use npm packages)'
```

Comment:
Nit: this seems unrelated to the commit message: `ci: upgrade to actions/github-script@v6`

Reply:
Ah, bad commit grouping.  I've fixed that now.